packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -2760,6 +2780,360 @@ void main() {
     expect(controller.selection.extent.offset, 5);
   }, skip: isBrowser);
 
+  testWidgets('keyboard text selection works as expected', (WidgetTester tester) async {
+    // Text with two separate words to select.
+    const String testText = 'Now is the time for\n'
+        'all good people\n'
+        'to come to the aid\n'
+        'of their country.';
+    final TextEditingController controller = TextEditingController(text: testText);
+    controller.selection = const TextSelection(
+      baseOffset: 0,
+      extentOffset: 0,
+      affinity: TextAffinity.upstream,
+    );
+    TextSelection selection;
+    SelectionChangedCause cause;
+    await tester.pumpWidget(MaterialApp(
+      home: Align(
+        alignment: Alignment.topLeft,
+        child: SizedBox(
+          width: 400,
+          child: EditableText(
+            maxLines: 10,
+            controller: controller,
+            showSelectionHandles: true,
+            autofocus: true,
+            focusNode: FocusNode(),
+            style: Typography(platform: TargetPlatform.android).black.subhead,
+            cursorColor: Colors.blue,
+            backgroundCursorColor: Colors.grey,
+            selectionControls: materialTextSelectionControls,
+            keyboardType: TextInputType.text,
+            textAlign: TextAlign.right,
+            onSelectionChanged: (TextSelection newSelection, SelectionChangedCause newCause) {
+              selection = newSelection;
+              cause = newCause;
+            },
+          ),
+        ),
+      ),
+    ));
+
+    await tester.pump(); // Wait for autofocus to take effect.
+
+    Future<void> sendKeys(List<LogicalKeyboardKey> keys, {bool shift = false, bool control = false}) async {
```

Comment:
I was wondering if we had a good way to test hardware keyboard keys 👍

Reply:
Yes!  I added that stuff last month because it was missing. It's turning out to be very useful.